training/src/anemoi/training/diagnostics/logger.py

```diff
@@ -27,79 +26,33 @@ def get_mlflow_logger(config: BaseSchema) -> None:
-
-    os.environ["MLFLOW_HTTP_REQUEST_MAX_RETRIES"] = str(http_max_retries)
-    os.environ["_MLFLOW_HTTP_REQUEST_MAX_RETRIES_LIMIT"] = str(http_max_retries + 1)
-    # these are the default values, but set them explicitly in case they change
-    os.environ["MLFLOW_HTTP_REQUEST_BACKOFF_FACTOR"] = "2"
-    os.environ["MLFLOW_HTTP_REQUEST_BACKOFF_JITTER"] = "1"
-
-    from anemoi.training.diagnostics.mlflow import LOG_MODEL
-    from anemoi.training.diagnostics.mlflow import MAX_PARAMS_LENGTH
-    from anemoi.training.diagnostics.mlflow.logger import AnemoiMLflowLogger
-
-    resumed = config.training.run_id is not None
-    forked = config.training.fork_run_id is not None
-
-    save_dir = config.hardware.paths.logs.mlflow
-
-    offline = config.diagnostics.log.mlflow.offline
-    if not offline:
-        tracking_uri = config.diagnostics.log.mlflow.tracking_uri
-        LOGGER.info("AnemoiMLFlow logging to %s", tracking_uri)
-    else:
-        tracking_uri = None
-
-    if (resumed or forked) and (offline):  # when resuming or forking offline -
-        # tracking_uri = ${hardware.paths.logs.mlflow}
-        tracking_uri = str(save_dir)
-    # create directory if it does not exist
-    Path(config.hardware.paths.logs.mlflow).mkdir(parents=True, exist_ok=True)
-
-    log_hyperparams = True
-    if resumed and not config.diagnostics.log.mlflow.on_resume_create_child:
-        LOGGER.info(
-            (
-                "Resuming run without creating child run - MLFlow logs will not update the"
-                "initial runs hyperparameters with those of the resumed run."
-                "To update the initial run's hyperparameters, set "
-                "`diagnostics.log.mlflow.on_resume_create_child: True`."
-            ),
-        )
-        log_hyperparams = False
-
-    max_params_length = getattr(config.diagnostics.log.mlflow, "max_params_length", MAX_PARAMS_LENGTH)
-    LOGGER.info("Maximum number of params allowed to be logged is: %s", max_params_length)
-    log_model = getattr(config.diagnostics.log.mlflow, "log_model", LOG_MODEL)
-
-    logger = AnemoiMLflowLogger(
-        experiment_name=config.diagnostics.log.mlflow.experiment_name,
-        project_name=config.diagnostics.log.mlflow.project_name,
-        tracking_uri=tracking_uri,
-        save_dir=save_dir,
-        run_name=config.diagnostics.log.mlflow.run_name,
+    logger_config = OmegaConf.to_container(convert_to_omegaconf(config.diagnostics.log.mlflow))
+    logger_config.pop("enabled")
+
+    # TODO: this seems a bit dirty
+    log_system = logger_config.pop("system")
+    log_terminal = logger_config.pop("terminal")
+    expand_hyperparams = logger_config.pop("expand_hyperparams")
+
+    # Defaults that exist outside the scope of config.diagnostics.log.mlflow
+    if "save_dir" in logger_config:
+        if logger_config["save_dir"] is None:
+            logger_config["save_dir"] = config.hardware.paths.logs.mlflow
```

Comment:
@anaprietonem I have this line in here since there was a version of it in the original code. But since `config.hardware.paths.logs.mlflow` is now in the default config yaml, do you think we can just remove this line (and nested if statements)?